x/trigger/simulation/genesis.go

```diff
@@ -0,0 +1,140 @@
+package simulation
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/provenance-io/provenance/internal/pioconfig"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+const (
+	TriggerID  = "trigger_id"
+	QueueStart = "queue_start"
+)
+
+// TriggerIDFn randomized MinActions
+func TriggerIDFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 2, 10000000000))
+}
+
+// QueueStartFn randomized MinActions
+func QueueStartFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 0, 10000000000))
+}
+
+// GetRandomEvent returns a random event
+func GetRandomEvent(r *rand.Rand, now time.Time) types.TriggerEventI {
+	rand := simtypes.RandIntBetween(r, 0, 1)
+	if rand > 0 {
```

Comment:
I know that `RandIntBetween`'s comment says that the min and max are inclusive, but they're not.

```golang
// RandIntBetween returns a random int between two numbers inclusively.
func RandIntBetween(r *rand.Rand, min, max int) int {
	return r.Intn(max-min) + min
}
```

The `r.Intn(n)` function returns a number between 0 (inclusive) and n (exclusive), i.e. `[0,n)`, i.e. `n` will never be a result.

So `RandIntBetween(r, 0, 1)` is doing `r.Intn(1-0) + 0`. But `r.Intn(1)` will always return `0`.

For this one, it's probably better to just use `r` directly, skip the variable assignment and just compare the result.

```suggestion
	if r.Intn(2) == 0 {
```